### PR TITLE
[B2BP-499, B2BP-504] - Modify-Workflow-DeployWebSite-for-Multiple-CDN-and-Tenant

### PR DESCRIPTION
**Pull request by maurodandrea**
**Ready on GitHub secrets/variables for the tenants "send" and "appio"**

Modify Workflow GitHub "DeployWebSite" for manage multiple CloudFront distribution and multi-Tenant
Depends on #238

This [Jira task](https://pagopa.atlassian.net/browse/B2BP-511?atlOrigin=eyJpIjoiYWE0NmJiOWZiMDZjNGYwN2JiOGI3YWU2OGIzOWJkNWQiLCJwIjoiaiJ9) shows the procedure to be carried out for reorganizing the github vars and secrets for multitenancy, subsequently it will be possible to add further values ​​to the environment in the workflow deploy_website code.

#### List of Changes
- Modify action Deploy for copy files in folder S3
- Modify Workflow DeployWebsite for envirnment tenant

#### Motivation and Context
Modify Workflow GitHub "DeployWebSite" for manage multiple CloudFront distribution and multi-Tenant

#### How Has This Been Tested?
On private GitHub repository

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [X] My change not requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/actions/deploy/action.yaml [hunk 1]
```diff
@@ -27,6 +27,9 @@ inputs:
   cdn_path:
     description: 'The CDN bucket S3 URL'
     required: false
+  tenant:
+    description: 'The Tenant inherent to the website'
+    required: false
```
Comment (kin0992) [inline .github/actions/deploy/action.yaml:32]: why is this field not required? what happen if a caller do not pass the `tenant` as inout?

Comment (maurodandrea) [reply .github/actions/deploy/action.yaml]: sorry. with last commit i set the parameter "required: true"

File: .github/workflows/deploy_website.yaml [hunk 2]
```diff
@@ -9,12 +9,11 @@ on:
   workflow_dispatch:
     inputs:
       environment:
-        description: 'The environment used as target'
+        description: 'The tenant environment used as target'
```
Comment (simosalsi) [inline .github/workflows/deploy_website.yaml:12]:
```suggestion
        description: 'The Tenant for which the static site will be built
```
I would keep the description consistent.

**Review by simosalsi — CHANGES_REQUESTED** [review]

**Review by simosalsi — APPROVED** [review]

File: .github/actions/deploy/action.yaml [hunk 3]
```diff
@@ -3,7 +3,7 @@ description: Deploy on AWS S3 Bucket
 
 inputs:
   environment:
```
Comment (kin0992) [inline .github/actions/deploy/action.yaml:5]: **question (non-blocking)**: why not change the input from `environment` to `tenant` then?

**Review by kin0992 — COMMENTED** [review]
**question**: are the environment variables set for the `send` tenant?

Approving this PR we are fetching environment variables from the `send` tenant, which is an `environment` here on GitHub; are we set everything up to let this workflow work?

File: .github/actions/deploy/action.yaml (hunk 3 above)
Comment (maurodandrea) [reply .github/actions/deploy/action.yaml:5]: Today we configured the env var on github with Christian for the tenant send test environment. The test has ended, so the env var will need to be modified with the new CDN values (for production). So I'm putting the PR in draft until everything is ready for the production environment.

The input was originally called tenant, then we decided in call to leave it with the name environment

**Review by christian-calabrese — APPROVED** [review]
LGTM

**Review by christian-calabrese — APPROVED** [review]

**Pull request merged**